docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst

```diff
@@ -0,0 +1,270 @@
+   writer
+-  A create method that instantiates a driver with that factory
+
+Concretely the interface is defined as the following in Scala:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/kms-driver-api/src/main/scala/com/digitalasset/canton/driver/api/v1/DriverFactory.scala
+   :language: Scala
+
+
+A specialization of the generic DriverFactory is the ``v1.KmsDriverFactory`` which
+defines the driver type to be a KmsDriver and the API version to be 1.
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/kms-driver-api/src/main/scala/com/digitalasset/canton/crypto/kms/driver/api/v1/KmsDriverFactory.scala
+   :language: Scala
+
+Driver Configuration Reading & Parsing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Canton uses `pureconfig <https://pureconfig.github.io/>`__ configuration library
+to read its configuration. Given that the configuration of the driver can be
+embedded inside a Canton configuration file, the factory of the driver needs to
+specify the type of configuration, as well as configuration readers and writers.
+A ``ConfigType`` can be a case class or as basic as a ``Map[String, String]``.
+In the latter case, the config reader and writer can be defined as:
+
+- ``val configReader = pureconfig.ConfigReader.mapReader[String]``
+- ``val configWriter = pureconfig.ConfigWriter.mapWriter[String]``
+
+KMS Driver API v1
+~~~~~~~~~~~~~~~~~
+
+The main part of the API is the ``v1.KmsDriver`` API that defines the following
+operations for a KMS Driver:
+
+-  Key Generation: Asymmetric signing and encryption key pairs as well
+   as symmetric encryption keys
+-  Supported Key and Algorithm Specifications: The specs that are
+   supported by the driver, for example, RSA 2048 keys and RSA OAEP
+   SHA256 asymmetric encryption.
+-  Signing: Sign data with a key from the KMS and the specified
+   algorithm.
+-  Asymmetric Decryption: Decrypt a ciphertext with a private key from
+   the KMS and the specified algorithm.
+-  Symmetric Encryption and Decryption: Encrypt or decrypt with a
+   symmetric encryption key from the KMS. The default symmetric
+   encryption algorithm of the KMS is used.
+-  Key Management: Get the public key of a private key stored in the
+   KMS, check if a key exists, and delete a key.
+-  Health: Return the health of the KMS Driver instance.
+
+The API is designed as an asynchronous API using Futures. An
+OpenTelemetry trace context is passed in from Canton into the KMS driver
+operations to be able to link a Canton request to operations in the KMS.
+The driver is responsible for propagating the trace context into the
+KMS.
+
+Concretely the Scala interface is defined as the following:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/kms-driver-api/src/main/scala/com/digitalasset/canton/crypto/kms/driver/api/v1/KmsDriver.scala
+   :language: Scala
+
+Error Handling and Health
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case the driver experiences an error the Future of the operation should be
```

Comment:
should fail

Reply:
actually the future should be completed as a failure with the exception, so I think `should be failed` is correct here.